Lux/src/fimage.cpp

```diff
@@ -0,0 +1,69 @@
+// Floating point image class using image template as a base
+
+#include "fimage.hpp"
+#include <memory>
+#include "image_loader.hpp"
+
+// pixel modification functions
+void fimage :: grayscale() {
+    std :: transform( base.begin(), base.end(), base.begin(),  []( frgb &f ) { return gray( f ); } );
+}
+
+void fimage :: load( const std :: string& filename ) {
+    reset();
+    image_loader loader( filename );
+    set_dim( { loader.xsiz, loader.ysiz } );
+
+	int c = 0;
+    frgb f;
+
+    for (auto it = begin (loader.img); it <= end (loader.img); ) {
+        if( loader.channels == 1 )	// monochrome image
+        {
+            setrc( f, *it );
+            setgc( f, *it );
+            setbc( f, *it );
+            it++;
+        }
+
+        if( loader.channels == 2 ) // don't know what to do in this case
+        {
+            setrc( f, *it );
+            setgc( f, *it );
+            setbc( f, *it );
+            it++; it++;
+        }
+
+        if( ( loader.channels == 3 ) | ( loader.channels == 4 ) )
+        {
+            setrc( f, *it );
+            it++;
+            setgc( f, *it );
+            it++;
+            setbc( f, *it );
+            it++;
+        }
+
+        // skip alpha channel - rgba ... if argb need to move line up
+        if( loader.channels == 4 ) it++;	
+        base.push_back( f );
+    }
+    mip_it();
+}
+
+void fimage :: quantize( std :: vector< unsigned char >& img )
+{
+    std :: for_each( base.begin(), base.end(), [ &img ]( frgb &f ) {
```

Comment:
i think frgb &f can be const and probably should be for increased readability

Reply:
Fixed locally. Will update branch when I've worked through all the comments.